src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
+const form = document.querySelector('form');
 
 function waitFor(element, eventName) {
-  // write your code here
+  if (element) {
+    return new Promise((resolve, reject) => {
+      element.addEventListener(eventName, () => {
+        const string = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
```

Comment:
The string that is resolved in the Promise doesn't provide any useful information for the user. It would be more useful to resolve the Promise with the event object itself, so that the user can access any properties of the event that they need.